src/code/templates/clutch-game.js

```diff
@@ -495,45 +498,23 @@ export default class ClutchGame extends Component {
     return [shuffle(fatherPool), shuffle(motherPool)];
   }
 
-  static authoredDrakesToDrakeArray = function(authoredChallenge, trial) {
-    const mother = new BioLogica.Organism(BioLogica.Species.Drake,
-                                          authoredChallenge.mother.alleles,
-                                          authoredChallenge.mother.sex),
-          father = new BioLogica.Organism(BioLogica.Species.Drake,
-                                          authoredChallenge.father.alleles,
-                                          authoredChallenge.father.sex),
-          // authored specs may be incomplete; these are complete specs
-          motherSpec = { alleles: mother.getAlleleString(), sex: mother.sex },
-          fatherSpec = { alleles: father.getAlleleString(), sex: father.sex };
-
-    // already generated drakes
-    if (trial > 0)
-      return authoredDrakes;
-
-    // challengeType === 'match-target'
-    targetDrakes = authoredChallenge.targetDrakes;
-    const targetDrakeCount = targetDrakes.length;
-
-    function childDrakesContain(alleles) {
-      for (let i = 3; i < authoredDrakes.length; ++i) {
-        if (authoredDrakes[i].alleles === alleles)
-          return true;
-      }
-      return false;
-    }
-
-    authoredDrakes = [motherSpec, fatherSpec];
-    for (let i = 0; i < targetDrakeCount; ++i) {
-      let child, childAlleles;
-      do {
-        child = BioLogica.breed(mother, father, false);
-        childAlleles = child.getAlleleString();
-        // don't generate the same set of alleles twice
-      } while (childDrakesContain(childAlleles));
-
-      authoredDrakes.push({ alleles: childAlleles, sex: child.sex });
+  static authoredDrakesToDrakeArray = function(authoredChallenge, trial, unscrambledTrialNumber) {
+    const changeableDrake = new BioLogica.Organism(BioLogica.Species.Drake,
+                                                   authoredChallenge.changeableDrakes[trial].alleles,
+                                                   BioLogica.FEMALE),
+          staticDrake = new BioLogica.Organism(BioLogica.Species.Drake,
+                                               authoredChallenge.staticDrakes[trial].alleles,
+                                               BioLogica.FEMALE);
+    changeableDrakeType = authoredChallenge.changeableDrakeType[unscrambledTrialNumber];
+    let fatherSpec, motherSpec;
+    if (changeableDrakeType === "mother") {
+      motherSpec = { alleles: changeableDrake.getAlleleString(), sex: BioLogica.FEMALE };
+      fatherSpec = { alleles: staticDrake.getAlleleString(), sex: BioLogica.MALE };
+    } else {
+      motherSpec = { alleles: staticDrake.getAlleleString(), sex: BioLogica.FEMALE };
+      fatherSpec = { alleles: changeableDrake.getAlleleString(), sex: BioLogica.MALE };
     }
-    return authoredDrakes;
+    return [motherSpec, fatherSpec, authoredChallenge.targetDrakes[trial]];
```

Comment:
This adds a lot of complexity to this template, and I bet that we're going to want similar things in other layouts (the general idea of having different static genes between mother and father, maybe based on trial number), and it would be a shame to reproduce all this in those ones. It also relies on the side-effect of the `authoredDrakesToDrakeArray` function setting an instance variable `changeableDrakeType`, which seems sketchy -- it seems like we could easily call things out of order and have that not get set.

I think this could maybe remain for now, but the moment we find ourselves wanting the same feature elsewhere we should replace it with something more generic.

Currently we have the idea of `userChangeableGenes` and `visibleGenes`, which is not being used here because it applies to the whole template. It seems as if it wouldn't be too hard to expand that to be applied to named drakes

e.g.

```
visibleGenes: "wings legs horns",
userChangeableGenes: {
   mother: "wings, legs",
   father: ""
}
```

and then expanded further to be applied to trial number:

```
visibleGenes: "wings legs horns",
userChangeableGenes: [{
   mother: "wings, legs",
   father: ""
  },
  {
   mother: "",
   father: "horns"
  }
]
```

it would still need to be implemented at the template-level, because only the templates are aware of the names, but I think it would be a one-line addition to the templates that cared (i.e. instead of passing `userChangeableGenes` to the genomeView, they pass `userChangeableGenes[name]` (they would have already been passed in the correct trial in the array as props).